lib/ecto/query/builder/filter.ex

```diff
@@ -74,5 +78,23 @@ defmodule Ecto.Query.Builder.Filter do
     Builder.escape(expr, :boolean, %{}, vars, env)
   end
 
-  defp to_field(field), do: Macro.escape {{:., [], [{:&, [], [0]}, field]}, [], []}
+  def runtime!(_kind, []) do
+    true
+  end
+
+  def runtime!(kind, params) when is_list(params) do
+    params
+      |> Enum.map(fn
+        {field, value} when is_atom(field) ->
+          value = %Ecto.Query.Tagged{tag: nil, type: {0, field}, value: value}
+          {:==, [], [to_field(field), value]}
+        {_, _acc} ->
+          raise(ArgumentError, message: "expected a keyword list in #{kind}, " <>
```

Comment:
```elixir
raise ArgumentError, "expected a keyword list in #{kind}, got: `#{inspect params}`"
```

Rationale: 

1. for `ArgumentError` (and all exceptions since 1.1), we can pass a binary (and it will become message: "message")
2. No need for `Macro.to_string(params)` because now we are working with values and no AST. So inspect will do it. :)